### PR TITLE
Typescript migration of client/actions and some more components

### PR DESCRIPTION
**Pull request by mickmister**

### CODE REVIEW

File: webapp/src/components/github_label_selector/index.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+import {connect} from 'react-redux';
+import {Dispatch, bindActionCreators} from 'redux';
+
+import {getLabelOptions} from '../../actions';
+
+import GithubLabelSelector from './github_label_selector';
+
+const mapDispatchToProps = (dispatch: Dispatch) => ({
+    actions: bindActionCreators({getLabelOptions}, dispatch),
+}) as unknown as Actions;
```
Comment (raghavaggarwal2308) [inline webapp/src/components/github_label_selector/index.ts:13]: Is this correct to use `unknown` here?

File: webapp/src/components/issue_attribute_selector.tsx [hunk 2]
```diff
@@ -95,29 +107,31 @@ export default class IssueAttributeSelector extends PureComponent {
...
                 <ReactSelect
-                    {...this.props}
+                    {...props}
                     isClearable={true}
                     placeholder={'Select...'}
                     noOptionsMessage={() => noOptionsMessage}
                     closeMenuOnSelect={!this.props.isMulti}
                     hideSelectedOptions={this.props.isMulti}
-                    onChange={this.onChange}
+                    onChange={this.onChange as any}
```
Comment (raghavaggarwal2308) [inline webapp/src/components/issue_attribute_selector.tsx:130]: Is it possible to add a proper type here and on line 134 ?

**Review by raghavaggarwal2308 — CHANGES_REQUESTED** [review]

**Comment by wiggin77** [comment]
@raghavaggarwal2308 this is a large PR that is out of date.  I think we should close it and replace with smaller PRs that convert fewer components at a time.   Thoughts?

**Comment by raghavaggarwal2308** [comment]
@wiggin77 Sounds good, closing it for now, will create smaller PRs.

**Pull request closed without merging**